File: SumRtoLeaf.java
class Solution {
    int sum = 0;
    public int sumNumbers(TreeNode root) {
        if (root==null){
            return 0;
        }


        helper(root,0);
        return sum;
    }

    void helper(TreeNode root,int num){
        num = num*10+root.val;
        if (root.left!=null){
            helper(root.left,num);
        }
        if (root.right!=null){
            helper(root.right,num);
        }

        if (root.right==null&&root.left==null){
            sum+=num;
        }

    }

}
SC:O(n)
TC:O(n)

File: sumrtoleaf.cpp
// TC:O(N)
// SC:O(H)

// Approach:
// Here using post oreder traversal,
// can be solved any traversal we are following for each node we are passing sum

class Solution {
public:
int T_sum=0;
    int sumNumbers(TreeNode* root) {
         
        if(root==nullptr) return 0;
        BST(root,0); 
        return T_sum;    
    }

     void BST(TreeNode* root,int sum)
     {
         if(root==nullptr) return;

        BST(root->left,sum * 10 + root->val);
        BST(root->right,sum * 10 + root->val);
         if(root->left == nullptr && root->right== nullptr)
         {
             sum=sum*10+root->val;
             T_sum+=sum;
             return;
         }

        
      
     }
};